assemblies/assembly-planner-configuration.adoc

```diff
@@ -0,0 +1,1986 @@
+  </category>
+
+  ...
+
+</log4j:configuration>
+----
+
+. Optional: If you have a multitenant application where multiple `Solver` instances might be running at the same time, separate the logging of each instance into separate files:
++
+COMMENT: this section is about LOG4J so can we have a LOG4J example here?
++
+.. Surround the `solve()` call with  http://logback.qos.ch/manual/mdc.html[Mapped Diagnostic Context] (MDC):
++
+[source,java,options="nowrap"]
+----
+        MDC.put("tenant.name",tenantName);
+        MySolution bestSolution = solver.solve(problem);
+        MDC.remove("tenant.name");
+----
+
+.. Configure your logger to use different files for each ``${tenant.name}``.
+For example in Logback, use a `SiftingAppender` in ``logback.xml``:
++
+[source,xml,options="nowrap"]
+----
+  <appender name="fileAppender" class="ch.qos.logback.classic.sift.SiftingAppender">
+    <discriminator>
+      <key>tenant.name</key>
+      <defaultValue>unknown</defaultValue>
+    </discriminator>
+    <sift>
+      <appender name="fileAppender.${tenant.name}" class="...FileAppender">
+        <file>local/log/optaplanner-${tenant.name}.log</file>
+        ...
+      </appender>
+    </sift>
+  </appender>
+----
+
+
+
+[[randomNumberGenerator]]
+=== Configuring the random number generator
+
+Many heuristics and metaheuristics depend on a pseudorandom number generator for move selection, to resolve score ties, probability based move acceptance, and so on. During solving, the same `Random` instance is reused to improve reproducibility, performance, and uniform distribution of random values.
+
+A random seed is a number used to initialize a pseudorandom number generator.
+
+.Procedure
+. Optional: To change the random seed of that `Random` instance, specify a ``randomSeed``:
++
+[source,xml,options="nowrap"]
+----
+<solver xmlns="https://www.optaplanner.org/xsd/solver" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="https://www.optaplanner.org/xsd/solver https://www.optaplanner.org/xsd/solver/solver.xsd">
+  <randomSeed>0</randomSeed>
+  ...
+</solver>
+----
+
+. Optional: To change the pseudorandom number generator implementation, specify a value for the ``randomType`` property,where `_RANDOM_NUMBER_GENERATOR_` is a pseudorandom number generator listed below:
++
+COMMENT: Is this the configuration file?
++
+[source,xml,options="nowrap"]
+----
+<solver xmlns="https://www.optaplanner.org/xsd/solver" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="https://www.optaplanner.org/xsd/solver https://www.optaplanner.org/xsd/solver/solver.xsd">
+  <randomType>_RANDOM_NUMBER_GENERATOR_</randomType>
+  ...
+</solver>
+----
++
+The following pseudorandom number generator are supported:
+
+* `JDK` (default): Standard implementation (``java.util.Random``).
+* ``MERSENNE_TWISTER``: Implementation by http://commons.apache.org/proper/commons-math/userguide/random.html[Commons Math].
```

Comment:
Implementation of _RANDOM_NUMBER_GENERATOR_

Reply:
Here I meant that in comparison to community version MERSENNE_TWISTER were replaced by _RANDOM_NUMBER_GENERATOR_  and it is fine - it is hard to guess now that MERSENNE_TWISTER is the parameter we can use as _RANDOM_NUMBER_GENERATOR_